catroid/src/org/catrobat/catroid/stage/StageActivity.java

```diff
@@ -189,16 +180,53 @@ public void pause() {
 
 	public void resume() {
 		stageListener.menuResume();
-		FlashUtil.resumeFlash();
-		VibratorUtil.resumeVibrator();
+		resumeResources();
+	}
+
+	public void resumeResources() {
+		int requiredResources = ProjectManager.getInstance().getCurrentProject().getRequiredResources();
+		List<Sprite> spriteList = ProjectManager.getInstance().getCurrentProject().getSpriteList();
+
 		SensorHandler.startSensorListener(this);
-		FaceDetectionHandler.resumeFaceDetection();
 
-		CameraManager.getInstance().resumePreviewAsync();
+		for (Sprite sprite : spriteList) {
+			if (sprite.getPlaySoundBricks().size() > 0) {
+				stageAudioFocus.requestAudioFocus();
+			}
+		}
+
+		if ((requiredResources & Brick.CAMERA_FLASH) > 0) {
+			FlashUtil.resumeFlash();
+		}
 
-		ServiceProvider.getService(CatroidService.BLUETOOTH_DEVICE_SERVICE).start();
+		if ((requiredResources & Brick.VIBRATOR) > 0) {
+			VibratorUtil.resumeVibrator();
+		}
 
-		if (nfcAdapter != null) {
+		if ((requiredResources & Brick.FACE_DETECTION) > 0) {
+			FaceDetectionHandler.resumeFaceDetection();
+		}
+
+		if ((requiredResources & Brick.BLUETOOTH_LEGO_NXT) > 0
+				|| (requiredResources & Brick.BLUETOOTH_PHIRO) > 0
+				|| (requiredResources & Brick.BLUETOOTH_SENSORS_ARDUINO) > 0) {
+			ServiceProvider.getService(CatroidService.BLUETOOTH_DEVICE_SERVICE).start();
+		}
+
+		if ((requiredResources & Brick.CAMERA_BACK) > 0) {
```

Comment:
could you merge this line with line 220 like you did in line 210?

`if ((requiredResources & Brick.CAMERA_BACK) > 0 || (requiredResources & Brick.CAMERA_FRONT) > 0)`